GoodThingsComein3s/ViewControllers/DetailsViewController.m

```diff
@@ -0,0 +1,144 @@
+//
+//  DetailsViewController.m
+//  GoodThingsComein3s
+//
+//  Created by Ruhee Rajwani on 7/28/22.
+//
+
+#import "DetailsViewController.h"
+#import "APIManager.h"
+#import "AFNetworking.h"
+#import "RestaurantPhotoCollectionViewCell.h"
+
+@interface DetailsViewController () <UICollectionViewDelegate,UICollectionViewDataSource, UICollectionViewDelegateFlowLayout>
+
+@property (weak, nonatomic) IBOutlet UILabel *restaurantNameLabel;
+@property (weak, nonatomic) IBOutlet UILabel *restaurantPriceLabel;
+@property (weak, nonatomic) IBOutlet UILabel *restaurantCategoriesLabel;
+@property (weak, nonatomic) IBOutlet UILabel *restaurantAddressLabel;
+@property (weak, nonatomic) IBOutlet UICollectionView *restaurantPhotosCollectionView;
+@property (weak, nonatomic) IBOutlet UIImageView *restaurantRatingImageView;
+@property (nonatomic) NSDictionary *restaurantToShow;
+@property (nonatomic) NSMutableArray *imageURLS;
+@property (weak, nonatomic) IBOutlet UIPageControl *restaurantImageCollectionViewPageControl;
+@property (nonatomic) int currentIndex;
+@property (nonatomic) NSTimer *timer;
+
+@end
+
+@implementation DetailsViewController
+
+- (void) viewDidLoad {
+    [super viewDidLoad];
+    
+    self.restaurantPhotosCollectionView.delegate = self;
+    self.restaurantPhotosCollectionView.dataSource = self;
+
+    self.imageURLS = [[NSMutableArray alloc] init];
+    
+    [[APIManager shared] getRestaurantDetails:self.yelpRestaurantID completion:^(NSDictionary * _Nonnull restaurant, NSError * _Nonnull error) {
+        if(error == nil){
+        self.restaurantToShow = restaurant;
+        [self setView];
+        }
+    }];
+}
+
+-(void) setView {
```

Comment:
This method could be more readable by reusing the `Restaurant` class as mentioned above!